src/libraries/Common/src/Interop/OSX/Interop.libproc.cs

```diff
@@ -120,7 +120,14 @@ internal static unsafe int[] proc_listallpids()
         {
             // Get the number of processes currently running to know how much data to allocate
             int numProcesses = proc_listallpids(null, 0);
-            if (numProcesses <= 0)
+            if (numProcesses == 0 && IsAppSandboxEnabled())
```

Comment:
What is the error code (`Marshal.GetLastPInvokeError`) that the `proc_listallpids` returns when app is sandboxed? Would it better to check for that error code instead?

Reply:
The error code is `1` and I can definitely change the condition to check the code instead. The return value 0 is invalid because you there should always be at least 1 running process (self) and I thought this is enough to detect the failure.

Reply:
Error code 1 is `EPERM`. It makes sense. I think using `numProcesses == 0 && Marshal.GetLastPInvokeError() == EPERM` condition for the fallback would be better than the environment check.

Reply:
I modified the the condition according to your suggestion.